dss/dynamodb/__init__.py

```diff
@@ -84,6 +84,26 @@ def get_primary_key_items(*, table: str, key: str, return_key: str = 'body') ->
             yield item[return_key]['S']
 
 
+def get_all_key_attributes(*, table: str, hash_key: str, sort_key: str) -> typing.Dict:
```

Comment:
This seems to be duplicating `get_item()`.  Consider modifying the result of `get_item()` in `get_subscription()` for the subscription specific value.  If that cannot be done, consider modifying `get_item()` to be more generic.

Reply:
I do agree that the functionality it duplicated, but i dont think this PR is the place to refactor the code. Rather we can make a issue and submit another PR for that, I do think that the change needs  to occur, but the scope of this PR is for getting the subscriptions stats working. 

I think if we move forward with this technique we can ensure our PRs are easier to review.

Reply:
After looking at this a bit, I think you might want to look closer at what `get_item()` returns.  `get_all_key_attributes()` should be able to be removed and the values parsed from `get_item()` with no changes to `get_item()`.  There should be no refactor needed.

Reply:
```def get_item(*, table: str, hash_key: str, sort_key: Optional[str] = None, return_key: str = 'body') -> str:
    query = {'TableName': table,
             'Key': _format_item(hash_key=hash_key, sort_key=sort_key, value=None)}
    item = db.get_item(**query).get('Item')
    if item is None:
        raise DynamoDBItemNotFound(f'Query failed to fetch item from database: {query}')
    return item[return_key]['S']
``` 

get_item is hard coded to only work with a singular return_key that is passed as a parameter, and the value requested must have a datatype in the table set as String.

`get_all_key_attributes` will remove the ddb datatype context for us, no matter what type it is in the database.
```aws dynamodb get-item --table-name dss-subscriptions-v2-aws-dev --key '{"hash_key": {"S":"my_email_dawg }, "sort_key": {"S":"8188a98a-93b9-4044-9c7d-f3ecc1d8e38d" }}'
{
    "Item": {
        "sort_key": {
            "S": "8188a98a-93b9-4044-9c7d-f3ecc1d8e38d"
        },
        "hash_key": {
            "S": "my_email_dawg"
        },
        "successful": {
            "N": "1"
        },
        "attempts": {
            "N": "3"
        },
        "body": {
            "S": "{\"callback_url\": \"https://www.specterbyte.com\", \"jmespath_query\": \"event_type=='CREATE'\", \"owner\": \"my_email_dawg\", \"uuid\": \"8188a98a-93b9-4044-9c7d-f3ecc1d8e38d\", \"replica\": \"aws\"}"
        },
        "test": {
            "N": "2"
        }
    }
}``` get_item will only work for sort_key,hash_key,body.

Reply:
Alright, let's refactor this later.